### PR TITLE
refactor: deprecate `login_before_ready` in favor of `startup_script_behavior`

### LINKED ISSUE #7758: Add support for `startup_script_behavior` (replaces `login_before_ready`)
Add support for the new property implemented by https://github.com/coder/terraform-provider-coder/issues/124.

- [x] When set, `login_before_ready` takes priority over `startup_script_behavior`, `false => blocking`, `true => non-blocking`
- [x] Rename property in code where applicable

Ref: https://github.com/coder/coder/issues/7757

### PR DESCRIPTION
**Pull request by mafredri**
This is in many ways just a cosmetic change, but quite a lot of code was touched to make it.

When `startup_script_behavior` is set to a non-empty value, it is preferred, otherwise we attempt to read the `login_before_ready` value and translate it to `startup_script_behavior`.

Fixes #7758

Ref: #7757

### CODE REVIEW

File: provisionersdk/proto/provisioner.proto [hunk 1]
```diff
@@ -105,11 +105,12 @@ message Agent {
 	int32 connection_timeout_seconds = 11;
 	string troubleshooting_url = 12;
 	string motd_file = 13;
-	bool login_before_ready = 14;
+	// Field 14 was bool login_before_ready = 14, now removed.
```
Comment (mtojek) [inline provisionersdk/proto/provisioner.proto:108]: You can mark this field as `reserved`.

File: coderd/database/migrations/000125_rename_login_before_ready_to_startup_script_behavior.down.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE workspace_agents ADD COLUMN login_before_ready boolean NOT NULL DEFAULT TRUE;
```
Comment (mtojek) [inline coderd/database/migrations/000125_rename_login_before_ready_to_startup_script_behavior.down.sql:1]: nit: Do we need `begin` & `commit` around this block?

File: coderd/database/migrations/000125_rename_login_before_ready_to_startup_script_behavior.up.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+CREATE TYPE startup_script_behavior AS ENUM ('blocking', 'non-blocking');
```
Comment (mtojek) [inline coderd/database/migrations/000125_rename_login_before_ready_to_startup_script_behavior.up.sql:1]: nit: Do we need `begin` & `commit` around this block?

File: coderd/workspaceagents.go [hunk 4]
```diff
@@ -1126,7 +1126,7 @@ func convertWorkspaceAgent(derpMap *tailcfg.DERPMap, coordinator tailnet.Coordin
 		ConnectionTimeoutSeconds:     dbAgent.ConnectionTimeoutSeconds,
 		TroubleshootingURL:           troubleshootingURL,
 		LifecycleState:               codersdk.WorkspaceAgentLifecycle(dbAgent.LifecycleState),
-		LoginBeforeReady:             dbAgent.LoginBeforeReady,
```
Comment (mtojek) [inline coderd/workspaceagents.go:1129]: It is more like double-checking facts:

An old agent talking with the updated Coderd. Will the agent explode?

Comment (mafredri) [reply coderd/workspaceagents.go:1129]: Good check! It won't explode, but if, somehow, an old agent binary is started from a cache vs downloading the latest one, then the behavior could change from non-blocking to blocking. I'll add this field back to this struct and mark it deprecated.

Comment (mafredri) [reply coderd/workspaceagents.go:1129]: Fixed!

**Review by mtojek — APPROVED** [review]
Ship it!

**Pull request merged**